### PR TITLE
WIP tenant setup - Overall direction review

### PR DESCRIPTION
**Pull request by elv-wayne**
- Organized into separate files ElvSpace, ElvAccount with a new "**elv-admin**" command responsible for non Eluvio Live specific stuff.

### Single account creation, admin, contentAdmin, contentUsers groups and tenant depoly command:
```
eluvio space_tenant_create <tenant_name> <funds>

eg:
./eluvio space_tenant_create wayne-tenant-2 20
```

### Setting the consumers group in the tenant contract
```
 tenant_add_consumers <group_id> [addrs..]
 - Adds address(es) to the tenant's consumer group
```

### Checking if the addr is in the consumers group
```
tenant_has_consumer <group_id> <addr>     
- Returns true or false if addr is in the tenant consumer group
```

### Individual functions used in tenant_create:
```
  
account_create <funds> <account_name> [tenant_admins]                           
  - Create a new account -> mnemonic, address, private key
  
account_set_tenant <tenant_admins>        
  - Sets the tenant admins group address for this account.
  
account_show                              
  - Shows current account information.
  
group_create <name>                       
 - Create a new access group
  
group_add <group_address> <account_address> <is_manager>
 - Add account to access group
 
space_tenant_deploy <tenant_name> <owner_addr> <tenant_admin_addr>
  - Deploys a tenant contract
```

### CODE REVIEW

File: src/Config.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const net = "main"; // Set to "main" or "demov3"
+const net = "demov3"; // Set to "main" or "demov3"
```
Comment (elv-serban) [inline src/Config.js:1]: Make sure to revert to "main"

File: src/Config.js [hunk 2]
```diff
@@ -16,10 +16,14 @@ const consts = {
   main: {
     tokenUriStart: "https://main.net955305.contentfabric.io/s/main/q/",
     tokenUriEnd: "/meta/public/nft",
+    kmsAddress: "0xd962aff088ca845de83ce0db0c91b9a0b93d294f",
+    spaceAddress: "0x6612d94a31fab146b4c7ace60ddf3a1e5e40e500",
```
Comment (elv-serban) [inline src/Config.js:20]: space address can be derived from /config (can be removed in a separate PR)

File: eluvio [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#! /usr/bin/env bash
+
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+node $SOURCE_DIR/utilities/EluvioCli.js $*
```
Comment (elv-serban) [inline eluvio:5]: How about `elv-admin`?  (`eluvio` is too generic and we have been avoiding using the full name in tools)

File: src/Config.js (hunk 1 above)
Comment (elv-wayne) [reply src/Config.js]: fixed

File: eluvio (hunk 3 above)
Comment (elv-wayne) [reply eluvio]: changed to elv-admin.
Though eluvio would be an easy to remember and type "core" tool, but I'm ok with elv-admin.

File: src/Config.js (hunk 2 above)
Comment (elv-wayne) [reply src/Config.js:20]: ok will do next PR

**Review by elv-serban — APPROVED** [review]
Looks good

**Comment by elv-serban** [comment]
## Tenant setup
- tenant 'root key'   (funding can be done as space owner)
- tenant admins      (as space owner; transfer ownership to 'root key')
   (set 'tenant admins' address tenantId for root key wallet)
- tenant contract    (as space owner)
 
## Provisioning - media platform

Executed as tenant 'root key'

- groups
  - Content Admins
  - Content Users
- types
 - Title Master
 - Title
 - Title Collection
 - Index
- libraries
 - Properties
 - Title Masters
 - Title Mezzanines

## Provisioning - eluvio live

Executed as tenant 'root key'
Require media platform provisioning first

- types
  - NFT Template
  - NFT Collection
  - NFT Marketplace
  - NFT Drop Event
- provision minter key
- mint helper contract
- rights to auth service 'key'

**Comment by rd22-cell** [comment]
How do I get involved
